mne/viz/topomap.py

```diff
@@ -468,6 +474,35 @@ def plot_topomap(data, pos, vmin=None, vmax=None, cmap=None, sensors=True,
     from matplotlib.widgets import RectangleSelector
 
     data = np.asarray(data)
+
+    if isinstance(pos, Info):  # infer pos from Info object
+        picks = _pick_data_channels(pos)  # pick only data channels
+        pick_info(pos.copy(), picks)
+
+        # check if there is only 1 channel type, and n_chans matches the data
+        ch_type = set(channel_type(pos, idx)
+                      for idx, _ in enumerate(pos["chs"]))
+        info_help = ("Pick Info with e.g. ``mne.pick_info`` and "
+                     "``mne.channels.channel_indices_by_type``.")
+        if len(ch_type) > 1:
+            raise ValueError("Multiple channel types in Info structure. " +
+                             info_help)
+        elif len(pos["chs"]) != data.shape[0]:
+            raise ValueError("Number of channels in the Info object and "
+                             "the data array does not match. " + info_help)
+        else:
+            ch_type = ch_type.pop()
+
+        if any(type_ in ch_type for type_ in ('planar', 'grad')):
+            # deal with grad pairs
+            from ..channels.layout import (_merge_grad_data, find_layout,
+                                           _pair_grad_sensors)
+            picks, pos = _pair_grad_sensors(pos, find_layout(pos))
+            data = _merge_grad_data(data[picks]).squeeze()
+        else:
+            picks = range(data.shape[0])
```

Comment:
I'm converting this to be on the save side, but I exclusively use Python 3 and I send ranges to picks. It works.